homeassistant/components/skybell/__init__.py

```diff
@@ -98,83 +89,68 @@ async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
     )
     try:
         devices = await api.async_initialize()
-    except SkybellAuthenticationException as ex:
-        raise ConfigEntryAuthFailed(
-            f"Authentication Error: please check credentials: {ex}"
-        ) from ex
+    except SkybellAuthenticationException:
+        return False
     except SkybellException as ex:
         raise ConfigEntryNotReady(f"Unable to connect to Skybell service: {ex}") from ex
 
-    async def async_update_data() -> None:
-        """Fetch data from API endpoint."""
-        try:
-            [await device.async_update() for device in devices]
-        except SkybellException as err:
-            raise UpdateFailed(f"Failed to communicate with device: {err}") from err
-
-    coordinator = DataUpdateCoordinator(
-        hass,
-        logging.getLogger(__name__),
-        name=DEFAULT_NAME,
-        update_method=async_update_data,
-        update_interval=MIN_TIME_BETWEEN_UPDATES,
-    )
-    await coordinator.async_config_entry_first_refresh()
-    hass.data[DOMAIN][entry.entry_id] = {
-        DATA_COORDINATOR: coordinator,
-        DATA_DEVICES: devices,
-    }
-
+    device_coordinators: dict[str, DataUpdateCoordinator] = {}
+    for device in devices:
+        coordinator = SkybellDataUpdateCoordinator(
+            hass,
+            device,
+        )
+        await coordinator.async_config_entry_first_refresh()
+        device_coordinators[device.device_id] = coordinator
```

Comment:
I might just make this a list

Reply:
You could write it as 

```
    device_coordinators: dict[str, DataUpdateCoordinator] = {
        device.device_id: SkybellDataUpdateCoordinator(hass, device)
        for device in devices
    }
    await asyncio.gather(
        *[
            coordinator.async_config_entry_first_refresh()
            for coordinator in device_coordinators.values()
        ]
    )
```

Reply:
Hm, everything is unavailable when it is written like this.

Reply:
nvm